Add section Trust Domain Name Collisions (#126)
spiffe/spiffe: standards/SPIFFE-ID.md

The trust domain name is defined as the authority component of the URI - specifically, the `host` part of the authority. The `userinfo` and `port` parts of the authority component MUST NOT be set, and the `:` delimiter MUST NOT be present. Please see section 3.2 of [RFC 3986](https://tools.ietf.org/html/rfc3986) for more information.

#### 2.1.1. Trust Domain Name Collisions

> Trust domain names are self-registered, ie. trust domain operators are free to choose any trust domain name they find suitable. Since there is no centralized authority for regulation or registration of trust domain names, there is no guarantee of global uniqueness nor is there any technical means for preventing distinct trust domains from using identical trust domain names.

Reviewer: ```suggestion
Trust domain operators are free to choose any trust domain name they find suitable: there is no centralized authority for regulation or registration of trust domain names. Thus, there is no guarantee of global uniqueness nor is there any technical means for preventing distinct trust domains from using identical trust domain names.
```

"self-registration" normally means there is a central authority, but the registration process is automated or unattended, like a sign-up sheet posted on a bulletin board, or creating a Twitter account.  Trust domain names aren't like that, so I've dropped that phrase and reworked things a bit. PTAL

> To prevent accidental collisions (two trust domains select identical names), operators are advised to select trust domain names which are highly likely to be globally unique. When available, using a registered domain name as a suffix of a trust domain name will reduce chances of an accidental collision; for example, if a trust domain operator owns the domain name `example.com`, then using a trust domain name such as `trust_domain_name.example.com` would likely not produce a collision. When trust domain names are automatically generated without operator input, randomly generating a unique name (such as a UUID) is strongly advised.

Reviewer 2: Here's the small suggestion we talked about in SIG-SPEC. I didn't end up changing the language around random generation of the trust domain name. After sleeping on it, the current language feels ok to me.
```suggestion
To prevent accidental collisions (two trust domains select identical names), operators are advised to select trust domain names which are highly likely to be globally unique. Even though a trust domain name is not a DNS name, using a registered domain name as a suffix of a trust domain name, when available, will reduce chances of an accidental collision; for example, if a trust domain operator owns the domain name `example.com`, then using a trust domain name such as `trust_domain_name.example.com` would likely not produce a collision. When trust domain names are automatically generated without operator input, randomly generating a unique name (such as a UUID) is strongly advised.
```

> To prevent accidental collisions (ie. two trust domains select identical names), operators are advised to select trust domain names which are highly likely to be globally unique. When available, using a registered domain name as a suffix of a trust domain name will reduce chances of an accidental collision; for example, if a trust domain operator owns the domain name `example.com`, then using a trust domain name such as `trust_domain_name.example.com` would likely not produce a collision. When trust domain names are automatically generated without operator input, random generating a unique name (such as a UUID) is strongly advised.

Reviewer: ```suggestion
To prevent accidental collisions (two trust domains select identical names), operators are advised to select trust domain names which are highly likely to be globally unique. When available, using a registered domain name as a suffix of a trust domain name will reduce chances of an accidental collision; for example, if a trust domain operator owns the domain name `example.com`, then using a trust domain name such as `trust_domain_name.example.com` would likely not produce a collision. When trust domain names are automatically generated without operator input, random generating a unique name (such as a UUID) is strongly advised.
```
i.e. isn't needed since you have parenthesis

> When a collision does occur, those trust domains will continue to operate independently but will be unable to federate (i.e. connect to one and other).  Because each trust domain uses unique cryptographic roots of trust, identity claims issued by one trust domain will fail validation in the other. Further details of SPIFFE authentication are covered in [Section 3.1](#31-svid-trust).

Reviewer: ```suggestion
When a collision does occur, those trust domains will continue to operate independently but will be unable to federate (connect to one another).  Because each trust domain uses unique cryptographic roots of trust, identity claims issued by one trust domain will fail validation in the other. Further details of SPIFFE authentication are covered in [Section 3.1](#31-svid-trust).
```

#### 2.1.1. Trust Domain Name Collisions

Trust domain names are self-registered, ie. trust domain operators are free to choose any trust domain name they find suitable. Since there is no centralized authority for regulation or registration of trust domain names, there is no guarantee of global uniqueness nor is there any technical means for preventing distinct trust domains from using identical trust domain names.

To prevent accidental collisions, operators are advised to select trust domain names which are highly likely to be globally unique. When available, using a registered domain name as a suffix of a trust domain name will reduce chances of an accidental collision; for example, if a trust domain operator owns the domain name `example.com`, then using a trust domain name such as `trust_domain_name.example.com` would be reasonable. When trust domain names are automatically generated without operator input, random generating a GUID is strongly advised.

> When a collision does occur (ie. two trust domains select identical names), those trust domains will continue to operate independently but will be unable to federate (ie. connect to one and other). Collisions, however, pose no impact to the security of a SPIFFE-compliant trust domain system: SPIFFE-based authenticatation relies on a (cryptographic) mechanism independent of a trust domain name. When performing SPIFFE-based authentication, the public key(s) of the root of a trust domain are used to validate an identity. Further details of SPIFFE authentication are covered in [Section 3.1](#31-svid-trust).

Reviewer: ```suggestion
When a collision does occur (ie. two trust domains select identical names), those trust domains will continue to operate independently but will be unable to federate (ie. connect to one and other). Collisions, however, pose no impact to the security of a SPIFFE-compliant trust domain system: identity claims issued by one trust domain will fail validation in the other and vice versa. Further details of SPIFFE authentication are covered in [Section 3.1](#31-svid-trust).
```

I feel like it's not completely accurate to say that authentication is "independent" of the trust domain name, since the trust domain name is used as a key to look up the correct trust bundle.  My suggestion is to skip the details entirely and say they'll be covered in a later section.

Author: I scrubbed most of the details. PTAL.
